tests/test_runtime_build.py

```diff
@@ -46,32 +46,41 @@ def test_vivado_hls():
     if os.system("which vivado_hls >> /dev/null") != 0:
         return 
 
-    hcl.init()
-    A = hcl.placeholder((10, 32), "A")
-    def kernel(A):
-        B = hcl.compute(A.shape, lambda *args : A[args] + 1, "B")
-        C = hcl.compute(A.shape, lambda *args : B[args] + 1, "C")
-        D = hcl.compute(A.shape, lambda *args : C[args] * 2, "D")
-        return D
-    
-    target = hcl.platform.aws_f1
-    s = hcl.create_schedule([A], kernel)
-    s.to(kernel.B, target.xcel)
-    s.to(kernel.C, target.host)
-    target.config(compile="vivado_hls", mode="sw_sim")
-    f = hcl.build(s, target)
-
-    np_A = np.random.randint(10, size=(10,32))
-    np_B = np.zeros((10,32))
-
-    hcl_A = hcl.asarray(np_A)
-    hcl_B = hcl.asarray(np_B, dtype=hcl.Int(32))
-    f(hcl_A, hcl_B)
-    ret_B = hcl_B.asnumpy()
-
-    for i in range(0, 10):
-      for j in range(0, 32):
-        assert ret_B[i, j] == (np_A[i, j] + 2) *2
+    def test_hls(target_mode):
+        hcl.init()
+        A = hcl.placeholder((10, 32), "A")
+        def kernel(A):
+            B = hcl.compute(A.shape, lambda *args : A[args] + 1, "B")
+            C = hcl.compute(A.shape, lambda *args : B[args] + 1, "C")
+            D = hcl.compute(A.shape, lambda *args : C[args] * 2, "D")
+            return D
+        
+        target = hcl.platform.aws_f1
+        s = hcl.create_schedule([A], kernel)
+        s.to(kernel.B, target.xcel)
+        s.to(kernel.C, target.host)
+        target.config(compile="vivado_hls", mode=target_mode)
+        f = hcl.build(s, target)
+
+        np_A = np.random.randint(10, size=(10,32))
+        np_B = np.zeros((10,32))
+
+        hcl_A = hcl.asarray(np_A)
+        hcl_B = hcl.asarray(np_B, dtype=hcl.Int(32))
+        f(hcl_A, hcl_B)
+        ret_B = hcl_B.asnumpy()
+
+        if "csyn" in target_mode:
+            report = f.report("csyn")
+            assert "ReportVersion" in report
+        elif "csim" in target_mode:
+            for i in range(0, 10):
+                for j in range(0, 32):
+                    assert ret_B[i, j] == (np_A[i, j] + 2) *2
```

Comment:
can you change it to  `np.testing.assert_array_equal(ret_B, (np_A+2)*2)`?

Reply:
Also comment out the GRU case to circumvent the testing error from keras: https://github.com/cornell-zhang/heterocl/blob/master/hlib/tests/test_keras.py#L320

Reply:
> can you change it to `np.testing.assert_array_equal(ret_B, (np_A+2)*2)`?

It seems all the test cases in `test_runtime_build.py` use `assert` instead of `np.testing.assert_array_equal`. Should I change all these cases?

Reply:
Yes. And for the Keras SegFault error, you can use the hard fix (i.e. comment of the GRU) first, just to pass the testing. I am working on the course project report, and will fix it soon before the weekend.

Reply:
Done.